### PR TITLE
Make validate_movie_writer actually check registered writers.

### PR DESCRIPTION
**Pull request by anntzer**
Right now the list in validate_movie_writer is outdated -- pillow is
missing from it.  Just use the actual list of registered writers instead
of trying to maintain it in sync.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/animation.py [hunk 1]
```diff
@@ -288,17 +288,17 @@ def __init__(self, fps=5, codec=None, bitrate=None, extra_args=None,
         self.frame_format = 'rgba'
 
         if codec is None:
-            self.codec = rcParams['animation.codec']
+            self.codec = mpl.rcParams['animation.codec']
```
Comment (anntzer) [inline lib/matplotlib/animation.py]: these changes are necessary to make animation importable from rcsetup.

Comment (timhoffm) [reply lib/matplotlib/animation.py]: Semi-OT: How can we reasonably handle the import hierarchy? Modules want to import `rcsetup` to have access to `rcParams` but `rcsetup` also wants to import some modules to read valid parameter values. I'm a bit concerned we'll run into cyclic imports sooner or later.

Comment (anntzer) [reply lib/matplotlib/animation.py]: In general I think doing `import matplotlib as mpl` and then accessing the rcParams as `mpl.rcParams` is good enough (as long as that access doesn't happen at module scope).

Comment (timhoffm) [reply lib/matplotlib/animation.py]: We then currently have the asymmetry, that some modules use that pattern while others use `from matplotlib import rcParams`. Should we switch all cases to `mpl.rcParams`? Importing a single variable into a module is anyway not too nice.

Comment (anntzer) [reply lib/matplotlib/animation.py]: I think they should all be switched to use mpl.rcParams to help with the circular import problem, but I don't think needs to be done *now*.

Comment (timhoffm) [reply lib/matplotlib/animation.py]: Agreed.

**Comment by anntzer** [comment]
(force-pushed to switch to use check_in_list as well)

**Comment by anntzer** [comment]
oops, looks like the force-push didn't go through.  Will do it later...

**Pull request merged**